scanner/src/funTest/kotlin/storages/AbstractStorageFunTest.kt

```diff
@@ -159,6 +160,20 @@ abstract class AbstractStorageFunTest : WordSpec() {
                 readResult should beSuccess()
                 (readResult as Success).result should beEmpty()
             }
+
+            "not store a result for the same scanner and provenance twice" {
+                val scanResult = ScanResult(provenanceWithSourceArtifact1, scannerDetails1, scanSummaryWithFiles)
+
+                val addResult1 = storage.add(id1, scanResult)
+                val addResult2 = storage.add(id1, scanResult)
+
+                addResult1 should beSuccess()
+                addResult2 should beFailure()
```

Comment:
I wonder if returning failure in that case is good, because from that failure the caller cannot know whether a
succeeding `read(id1)` would return the stored value. What do you think?

Also would overwriting make sense, like with `Map.put` ?

Reply:
> Makes sense, but on the other hand even though we expect scanners to be deterministic the new result might be different. It will at least have different timestamps. The current documentation of the add() function does not define the behavior for this situation. The reason I chose this implementation was that it was simpler, e.g. we don't have to check the SQL exception to determine the reason why the query has failed.

Makes sense. But still, would it make sense to keep the implementation of the storage as-is, but adjust the test so that it ensures that the second `add()` call does not overwrite anything? This would ensure that the different storage implementations behave in the same way.

> I wouldn't do that because it's less efficient to overwrite. For the file based storage we don't have to upload anything and for the Postgres storage we don't have to recover from the exception.

Make sense to me as well.